src/test/ui/pattern/pat-tuple-underfield.fixed

```diff
@@ -0,0 +1,51 @@
+// run-rustfix
+
+struct S(i32, f32);
+enum E {
+    S(i32, f32),
+}
+
+fn main() {
+    match S(0, 1.0) {
+        S(x, .., _) => {}
+        //~^ ERROR this pattern has 1 field, but the corresponding tuple struct has 2 fields
+        //~| HELP use `_` to explicitly ignore a field
+        //~| HELP use `..` to ignore all unmentioned fields
+    }
+    match S(0, 1.0) {
+        S(_, .., _) => {}
+        //~^ ERROR this pattern has 1 field, but the corresponding tuple struct has 2 fields
+        //~| HELP use `_` to explicitly ignore a field
+        //~| HELP use `..` to ignore all unmentioned fields
+    }
+    match S(0, 1.0) {
+        S(.._, _) => {}
+        //~^ ERROR this pattern has 0 fields, but the corresponding tuple struct has 2 fields
+        //~| HELP use `_` to explicitly ignore a field
+        //~| HELP use `..` to ignore all unmentioned fields
+    }
+
+    match E::S(0, 1.0) {
+        E::S(x, .., _) => {}
+        //~^ ERROR this pattern has 1 field, but the corresponding tuple variant has 2 fields
+        //~| HELP use `_` to explicitly ignore a field
+        //~| HELP use `..` to ignore all unmentioned fields
+    }
+    match E::S(0, 1.0) {
+        E::S(_, .., _) => {}
+        //~^ ERROR this pattern has 1 field, but the corresponding tuple variant has 2 fields
+        //~| HELP use `_` to explicitly ignore a field
+        //~| HELP use `..` to ignore all unmentioned fields
+    }
+    match E::S(0, 1.0) {
+        E::S(.._, _) => {}
```

Comment:
As you can see, unfortunately it is applying both suggestions. Is there any way to signal that only one should be applied by rustfix?

Reply:
I removed the test since it doesn't seem to be possible. I'm leaving this conversation unresolved in case someone knows a way to do it!

Reply:
This is a dual issue: we need to change rustc's json output to be more detailed (it cannot distinguish today between multiple one part suggestions and a single multipart suggestion) and rustfix needs to start consuming it. We started talking about this a year and a half ago, but it petered out, in part due to lack of time on my side. I'm hoping to rekindle this and start fixing this in the first half of 2021.

Reply:
That would be amazing! ❤️

Reply:
Ah, I did saw this. Did it break or it shows the incorrect behavior? If it's just incorrect behavior we could add it to the tests as well, and once that behavior is fixed we can go and fix the tests.